src/compiler/emitter.ts

```diff
@@ -2091,16 +2106,54 @@ namespace ts {
             return statements.length;
         }
 
+        function emitPrologueDirectivesIfNeeded(sourceFileOrBundle: Bundle | SourceFile) {
+            if (sourceFileOrBundle.kind === SyntaxKind.SourceFile) {
+                setSourceFile(sourceFileOrBundle as SourceFile);

```

Comment:
Do you need the casts? `Bundle | SourceFile` is a discriminated union, so the casts should not be necessary.